screensharing/main.js

```diff
@@ -85,6 +109,44 @@ class ScreenShareMainHook {
             case SCREEN_SHARE_EVENTS.STOP_SCREEN_SHARE:
                 this._jitsiMeetWindow.webContents.send(SCREEN_SHARE_EVENTS_CHANNEL, { data });
                 break;
+            case SCREEN_SHARE_EVENTS.DO_GDM: {
+                const { callback } = this._gdmData;
+
+                this._gdmData = null;
+
+                if (!data.streamId) {
+                    callback(null);
+
+                    break;
+                }
+
+                // The API requires a `DesktopCapturerSource` object. Find the matching one.
+                const options = {
+                    types: [ 'screen', 'window' ],
+                    thumbnailSize: {
+                        width: 0,
+                        height: 0
+                    },
+                    fetchWindowIcons: false
+                };
+
+                desktopCapturer.getSources(options).then(sources => {
```

Comment:
Am I correct that this should be the second `desktopCapturer.getSources(` call for the whole process - one from the desktop picker and this one?

Isn't it possible to avoid this?

Reply:
Yes, this is the 2nd one. The problem is we need the object, not the ID. I can try to cache them on the 1st call, but note this 2nd one doesn't fetch icons or thumbnails so it's much cheaper to run.

Reply:
I added a cache but still kept the fallback behavior in case they go out of sync. The window is very narrow, but let's err on the safe side.